src/bin/common/solving_loop.ml

```diff
@@ -778,6 +889,35 @@ let main () =
         handle_get_info st kind;
         st
 
+      | {contents = `Get_assignment; _} ->
+        begin
+          match State.get partial_model_key st with
+          | Some Model ((module SAT), partial_model) ->
+            begin
+              match SAT.get_model partial_model with
+              | Some _ ->
+                if State.get get_assignment st then
+                  handle_get_assignment
+                    ~get_value:(SAT.get_value partial_model)
+                    st
+                else
+                  recoverable_error
+                    "Produce assignments disabled; \
+                     add (set-option :produce-assignments true)";
+                st
+              | _ ->
+                recoverable_error
+                  "Model generation disactivated, cannot execute \
```

Comment:
`disabled`, not `deactivated`. We may want to mention `(set-option :produce-model true)` here.

Reply:
Although do we actually need this case? It looks like we don't care about the result of `SAT.get_model` and I think it should work without?

Reply:
The command should only be available in SAT mode, is that the good way to check it?